runbotics/runbotics-orchestrator-ui/src/src-app/components/tables/HistoryTable/HistoryTable.tsx

```diff
@@ -156,8 +156,36 @@ const HistoryTable = forwardRef<any, HistoryTableProps>(({ botId, processId, sx,
         setInstanceId(null);
         setPageSize(newPageSize);
     };
-    
-    const processInstanceColumns = useProcessInstanceColumns(rerunEnabled, handleRerunProcess);
+
+    const handleNoSubprocessesFound = (currRow: ProcessInstanceRow) => {
+        enqueueSnackbar(
+            translate('History.Table.Error.SubprocessesNotFound'),
+            { variant: 'error' },
+        );
+        dispatch(processInstanceActions.updateProcessInstance({ id: currRow.original.id, hasSubprocesses: false }));
+    };
+
+    const getSubprocessesCount = (currRow: ProcessInstanceRow) => {
+        dispatch(processInstanceActions.getSubprocessesCount({ processInstanceId: currRow.original.id }))
+            .catch(() => { handleNoSubprocessesFound(currRow); });
+    };
+
+    const getSubprocessesPage = ({ currRow, pageNum, size }: GetSubprocessesPageParams) => {
+        dispatch(processInstanceActions.getSubprocesses({ processInstanceId: currRow.original.id, page: pageNum, size }))
+            .then((response) => {
+                if((response as GetSubprocessesResponse).payload.length === 0) handleNoSubprocessesFound(currRow);
+
+                dispatch(processInstanceActions.getSubprocessesCount({ processInstanceId: currRow.original.id }));
+            })
+            .catch(() => { handleNoSubprocessesFound(currRow); });
+    };
+
+    const firstSubprocessesLoad = ({ currRow, pageNum, size }: GetSubprocessesPageParams) => {
+        getSubprocessesPage({ currRow, pageNum, size });
+        getSubprocessesCount(currRow);
+    };
```

Comment:
Line 178 and 185 - so we do it twice? Why?

Reply:
removed fn so it's outdated